File: code/stable/projectile/projectile_s_l_nol_u_u_v_f/src/csharp/InputConstraints.cs
/** \file InputConstraints.cs
    \author Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
    \brief Provides the function for checking the physical constraints on the input
*/
using System;

public class InputConstraints {
    
    /** \brief Verifies that input values satisfy the physical constraints
        \param v_launch launch speed: the initial speed of the projectile when launched (m/s)
        \param theta launch angle: the angle between the launcher and a straight line from the launcher to the target (rad)
        \param p_target target position: the distance from the launcher to the target (m)
    */
    public static void input_constraints(float v_launch, float theta, float p_target) {
        if (!(v_launch > 0.0f)) {
            Console.Write("Warning: ");
            Console.Write("v_launch has value ");
            Console.Write(v_launch);
            Console.Write(", but is suggested to be ");
            Console.Write("above ");
            Console.Write(0.0f);
            Console.WriteLine(".");
        }
        if (!(0.0f < theta && theta < Math.PI / 2.0f)) {
            Console.Write("Warning: ");
            Console.Write("theta has value ");
            Console.Write(theta);
            Console.Write(", but is suggested to be ");
            Console.Write("between ");
            Console.Write(0.0f);
            Console.Write(" and ");
            Console.Write(Math.PI / 2.0f);
            Console.Write(" ((pi)/(2))");
            Console.WriteLine(".");
        }
        if (!(p_target > 0.0f)) {
            Console.Write("Warning: ");
            Console.Write("p_target has value ");
            Console.Write(p_target);
            Console.Write(", but is suggested to be ");
            Console.Write("above ");
            Console.Write(0.0f);
            Console.WriteLine(".");
        }
    }
}

File: code/stable/projectile/projectile_s_l_nol_u_u_v_f/src/swift/InputConstraints.swift
/** InputConstraints.swift
    Provides the function for checking the physical constraints on the input
    - Authors: Samuel J. Crawford, Brooks MacLachlan, and W. Spencer Smith
*/
/** Verifies that input values satisfy the physical constraints
    - Parameter v_launch: launch speed: the initial speed of the projectile when launched (m/s)
    - Parameter theta: launch angle: the angle between the launcher and a straight line from the launcher to the target (rad)
    - Parameter p_target: target position: the distance from the launcher to the target (m)
*/
func input_constraints(_ v_launch: Float, _ theta: Float, _ p_target: Float) -> Void {
    if !(v_launch > 0.0) {
        print("Warning: ", terminator: "")
        print("v_launch has value ", terminator: "")
        print(v_launch, terminator: "")
        print(", but is suggested to be ", terminator: "")
        print("above ", terminator: "")
        print(0.0, terminator: "")
        print(".")
    }
    if !(0.0 < theta && Double(theta) < Double.pi / Double(2.0)) {
        print("Warning: ", terminator: "")
        print("theta has value ", terminator: "")
        print(theta, terminator: "")
        print(", but is suggested to be ", terminator: "")
        print("between ", terminator: "")
        print(0.0, terminator: "")
        print(" and ", terminator: "")
        print(Double.pi / Double(2.0), terminator: "")
        print(" ((pi)/(2))", terminator: "")
        print(".")
    }
    if !(p_target > 0.0) {
        print("Warning: ", terminator: "")
        print("p_target has value ", terminator: "")
        print(p_target, terminator: "")
        print(", but is suggested to be ", terminator: "")
        print("above ", terminator: "")
        print(0.0, terminator: "")
        print(".")
    }
}
